### PR TITLE
Sending email with SMTP

### LINKED ISSUE #447: Add a guide on sending email using SMTP rather than mailgun
I think it'd be great if we had example of a more common + generic method of sending emails rather than just one specific third party service. :)

### PR DESCRIPTION
**Pull request by lpil**
@jeregrine How does this read? :)

Closes #447

### CODE REVIEW

**Comment by jeregrine** [comment]
@lpil this is good.  Lets hold off so I can get some community feedback regarding Bamboo. But the overall content is good.

**Comment by lpil** [comment]
While I can't say much for the internals, the API is wonderful. Significantly better designed than the Mailgun library that we also recommend.

Is there another library that handles SMTP nicely?

File: bonus_guides/I_sending_email_with_smtp.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+  server: "smtp.domain",
+  port: 1025,
+  username: SYSTEM.get_env("SMTP_USERNAME"),
+  password: SYSTEM.get_env("SMTP_PASSWORD"),
+  tls: :if_available, # can be `:always` or `:never`
+  ssl: false, # can be `true`
+  retries: 1
+```
+
+These confirm variables will need to be set on the production servers, as well
```
Comment (akappen) [inline bonus_guides/I_sending_email_with_smtp.md]: These variables will need. . .

"confirm" must be a typo

File: bonus_guides/I_sending_email_with_smtp.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    |> subject("Welcome!")
+    |> text_body("Welcome to MyApp!")
+    |> put_text_layout({MyApp.LayoutView, "email.text"})
+  end
+
+  def welcome_html_email(email_address) do
+    email_address
+    |> welcome_text_email()
+    |> html_body("<strong>Welcome<strong> to MyApp!")
+    |> put_text_layout({MyApp.LayoutView, "email.html"})
```
Comment (akappen) [inline bonus_guides/I_sending_email_with_smtp.md]: Should this be put_html_layout?

File: bonus_guides/I_sending_email_with_smtp.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    |> from("us@example.com")
+    |> subject("Welcome!")
+    |> put_text_layout({MyApp.LayoutView, "email.text"})
+    |> render("welcome.text")
+  end
+
+  def welcome_html_email(email_address) do
+    email_address
+    |> welcome_text_email()
+    |> put_text_layout({MyApp.LayoutView, "email.html"})
```
Comment (akappen) [inline bonus_guides/I_sending_email_with_smtp.md]: Should this also be put_html_layout?

File: bonus_guides/I_sending_email_with_smtp.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
```
Comment (akappen) [inline bonus_guides/I_sending_email_with_smtp.md]: Should this be put_html_layout?

File: bonus_guides/I_sending_email_with_smtp.md [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+    applications: [:phoenix, :phoenix_html, :cowboy, :logger, :gettext,
+                  :phoenix_ecto, :postgrex, :bamboo]]
+end
+```
+
+
+### Configuration
+
+We'll also need to add our SMTP service details to `config/config.ex`. These
+can be got from your SMTP provider, so check your account on there.
```
Comment (akappen) [inline bonus_guides/I_sending_email_with_smtp.md]: These can be gotten from your SMTP provider, so check your account there.

**Comment by lpil** [comment]
Thanks @akappen for helping fix my rubbish typing :)

**Comment by lpil** [comment]
Hi @jeregrine, what's the status here? Bamboo continues to be actively developed, and has more github stars than the Mailgun library recommended in the other guide, and currently has almost as many downloads per day.

**Comment by lpil** [comment]
Ping @jeregrine, can we get an update please?

**Comment by Gazler** [comment]
@lpil Sorry it has been so long. I think the contents are really well written, however I am not sure what the stance should be on 3rd party libraries (including Mailgun).

I'll try and draw some attention to this issue for discussion.

**Comment by lpil** [comment]
I feel that with phoenix's more limited scope we will struggle to make useful guides that don't recommend the use of libraries for tasks like these.

**Comment by lpil** [comment]
Also thank you for the reply! :)

**Comment by Gazler** [comment]
@lpil Could you please resolve the merge conflicts on this and I'll merge it in?

**Comment by lpil** [comment]
There you go @Gazler !

**Comment by Gazler** [comment]
Thanks for your patience with this @lpil Sorry again that it took so long.

**Comment by lpil** [comment]
Glad it's in :)

**Pull request merged**